### DISCUSSION TITLE
Deploy magic to existing nginx server

### DISCUSSION THREAD

**Discussion by kikiimdev** [polterguy/magic#188] [Q&A] [answered]
Hi, i'am trying to deploy Magic to my existing nginx server,
On **docker-compose.yaml** file, the used port for forwarding is **80** and **443**

On my server port **80** and **443** is already used by Nginx, is there a way to solve this problem?
I'am kinda new on docker tho

**Comment by polterguy** [comment, marked as answer]
You can try to edit the ports, but even if you succeed you'd end up with _"funny URLs"_ where you'd have to use an alternative port to access the thing. However, the `docker-compose.yml` file also automatically creates a LetsEncrypt SSL certificate for you, which assumes port 80 to verify ownership of the site. Implying even if you succeed editing the port, you'd probably not get SSL working, and even if you get SSL working, you'd end up with _"sub-optimal"_ URLs ...

It might be you could configure a reverse proxy for the `Host` name you want to use for Magic, and forward this into an internal port you could expose in the `docker-compose.yml` file. The latter would probably be your best solution, as in using your existing nGinx server, add a location to it, set it up as a reverse proxy, etc. As to the exact mechanics of this, I'm not sure - Google ...?

**Reply by kikiimdev** [reply]
Thanks for the reply,
I ended up using the later options and yeah its working, thank you

**Reply by polterguy** [reply]
Great, keep us in the loop if you need more help :)
